core/server/worker/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -423,6 +424,28 @@ void closeUfsBlock(long sessionId, long blockId)
       throws BlockAlreadyExistsException, BlockDoesNotExistException, IOException,
       WorkerOutOfSpaceException;
 
+  /**
+   * Gets the block reader to read from Alluxio block or UFS block.
+   * This operation must be paired with {@link #cleanBlockReader(long, long, BlockReader)}.
+   *
+   * @param request the block read request
+   * @return a block reader to read data from
+   */
+  BlockReader getBlockReader(BlockReadRequest request) throws IOException,
+      BlockDoesNotExistException, InvalidWorkerStateException,
+      BlockAlreadyExistsException, WorkerOutOfSpaceException;
+
+  /**
+   * Cleans data reader and related blocks after using the block reader obtained
+   * from {@link #getBlockReader(BlockReadRequest)}.
+   *
+   * @param sessionId the session id which used for getting the block reader
+   * @param blockId the block id this block reader belongs to
+   * @param reader the to be cleaned block reader
+   */
+  void cleanBlockReader(long sessionId, long blockId, BlockReader reader)
```

Comment:
nit: better to put read as the first arg as this method is about `cleanBlockReader`

Reply:
Moved BlockReader to be the first parameter, thanks